releasenotes/notes/0.16/extend-updates-cached-qubits-2dbafe010bd5a946.yaml

```diff
@@ -0,0 +1,14 @@
+---
+fixes:
+  - |
+    Previously, extending :class:`~qiskit.circuit.QuantumCircuit` objects
+    using the :meth:`~qiskit.circuit.QuantumCircuit.extend` method with
+    a circuit that had differing register would lead to a discrepency in the
+    :attr:`~qiskit.circuit.QuantumCircuit.qubits`  and
+    :attr:`~qiskit.circuit.QuantumCircuit.clbits` attributes because the
+    cached lists were not updated. THis has been fixed so that calling
+    :meth:`~qiskit.circuit.QuantumCircuit.extend` updates
+    :attr:`~qiskit.circuit.QuantumCircuit.qubits` and
+    :attr:`~qiskit.circuit.QuantumCircuit.clbits` with any new bits from
+    new registers being added to the circuit.
+    Fixes `#5038 <https://github.com/Qiskit/qiskit-terra/issues/5038>`__
```

Comment:
This was backported in 0.15.2 ( https://github.com/Qiskit/qiskit-terra/pull/5041 ).

Reply:
Removed in: https://github.com/Qiskit/qiskit-terra/pull/5142/commits/f68e0ae1727a696e4d3ecb673f52dcf42bf940d8